src/clock.c

```diff
@@ -27,6 +28,13 @@ static void clock_interrupt( int i, int code )
 {
 	clicks++;
 	process_wakeup_all(&queue);
+
+	if (terminal_has_input())
+	{
+		char input[1024];
+		int n = terminal_read(input, 1024);
+		console_write(0, input, n, 0);
+	}
```

Comment:
Hmm, this isn't quite the right place to put this code.

One reason is that this is an interrupt handler, and so the sort of code that can actually be invoked is very limited, usually just bumping some global variables.  `process_wakeup` is coded very carefully to be interrupt safe, but not all functions are.  (Side issue: We should clearly label which functions are interrupt safe.)

The problem here is that you want the *keyboard* interrupt to trigger some interaction between the terminal device and the console.  So, this would be better in the keyboard module.  However, we don't have a user-level context to make that happen.